arch/sim/src/sim/up_idle.c

```diff
@@ -116,4 +116,8 @@ void up_idle(void)
 
   up_timer_update();
 #endif
+
+#ifdef CONFIG_SIM_HCISOCKET
```

Comment:
it's better to move bthcisock_loop before up_timer_update because up_timer_update will:
1.Update the timestamp to the next tick
2.May switch the execution to the high priority task.